### ISSUE TITLE
Consider moving Boot source out of advanced options

### ISSUE THREAD

**Issue by vjuranek** [jenkinsci/openstack-cloud-plugin#204]
From my experience, users who are not very familiar with the plugin are quite often confused by config, as they intuitively think that configure `Name` and `Label` is enough, assuming that name matches OS image name, don't check advanced options and thus don't configure boot source and then had hard time to understand where the problem is (usually thinking there's some problem with they OS image metadata or something similar). 
Maybe it would be more user friendly to move `Boot source` out of Advanced options as this is typically option which has to be configured most often.
(I have no strong opinion on it, this is just my observation and suggestion to consider)

**Comment by olivergondza** [comment]
Good point, there has to be a deeper though and some code adjustment to refine which of the options are truly mandatory and  I agree those should not be hidden under *Advanced*.

**Comment by alanfx** [comment]
Just to add to this issue, the error you get when you try to use the template `Missing imageRef attribute` is not very helpful. Improving this message to mention the `Boot source` option of the template would direct the new user to the appropriate place to fix the problem.

**Comment by olivergondza** [comment]
That was exactly my point, now request is sent to openstack and the message is coming from there which can make little sense to the plugin users. Once we handle missing options we can not only avoid sending the request we know is doomed to fail but we can provide a better message as well.

**Comment by alanfx** [comment]
OK, I understand your point now, @olivergondza

**Comment by olivergondza** [comment]
When I comes to rearranging the controls, I have decided no to pull anything out as they are scoped by descriptor and logical sections and I do want to keep it that way. I am adding less vague Advanced button label to help people realize there is more to do.

**Comment by alanfx** [comment]
@olivergondza is there any verification you could do on the configuration to check that any required parameters are supplied so there won't be errors in OpenStack? For example, if the image name is missing, then make sure that some other parameter is supplied? Just wondering

**Comment by olivergondza** [comment]
@alanfx, do you think we need more of a visual aid?
![validation](https://user-images.githubusercontent.com/206841/42941426-0fdb7a2c-8b5d-11e8-90fe-815f09344fc7.png)

What I intend to do here is making sure the user who ignores this get more meaningful error message without roundtrip to OpenStack.

**Comment by alanfx** [comment]
@olivergondza yes, I think this would help a lot. I think it would be better if required options were visible in the plugin configuration, but providing some kind of feedback about them when performing the configuration would help a lot, IMO. Thanks

**Comment by olivergondza** [comment]
@alanfx, what I attached is a screenshot of current implementation. Hence my question, do we need more?

**Comment by alanfx** [comment]
@olivergondza sorry, I think I misunderstood what you were showing in the screenshot. So the answer is "yes", I think you need more. As a user, if I can just supply the `Name` and `Label` and save the configuration, then this is the issue that @vjuranek mentions in the original description of the problem. I really think that moving `Boot source` out of the Advanced options would improve usability, even though I know that means rearranging the code.
